File: VB/WebSite/App_Code/MyObject.vb
Imports Microsoft.VisualBasic
Imports System
Imports DevExpress.Xpo

Public Class MyObject
	Inherits XPObject
	Public Sub New()
		MyBase.New()
	End Sub

	Public Sub New(ByVal session As Session)
		MyBase.New(session)
	End Sub

	Public Overrides Sub AfterConstruction()
		MyBase.AfterConstruction()
	End Sub

	Protected _Title As String
	Public Property Title() As String
		Get
			Return _Title
		End Get
		Set(ByVal value As String)
			SetPropertyValue(Of String)("Title", _Title, value)
		End Set
	End Property

	Protected _Message As String
	Public Property Message() As String
		Get
			Return _Message
		End Get
		Set(ByVal value As String)
			SetPropertyValue(Of String)("Message", _Message, value)
		End Set
	End Property

	Private _IsRead As Boolean
	Public Property IsRead() As Boolean
		Get
			Return _IsRead
		End Get
		Set(ByVal value As Boolean)
			SetPropertyValue("IsRead", _IsRead, value)
		End Set
	End Property
End Class

File: CS/WebSite/App_Code/MyObject.cs
using System;
using DevExpress.Xpo;

public class MyObject : XPObject {
    public MyObject()
        : base() { }

    public MyObject(Session session)
        : base(session) { }

    public override void AfterConstruction() {
        base.AfterConstruction();
    }

    protected String _Title;
    public String Title {
        get { return _Title; }
        set { SetPropertyValue<String>("Title", ref _Title, value); }
    }

    protected String _Message;
    public String Message
    {
        get { return _Message; }
        set { SetPropertyValue<String>("Message", ref _Message, value); }
    }

    private Boolean _IsRead;
    public Boolean IsRead
    {
        get { return _IsRead; }
        set { SetPropertyValue("IsRead", ref _IsRead, value); }
    }
}

